cel/canonical_eventlog.go

```diff
@@ -117,26 +124,68 @@ type CEL struct {
 	Records []Record
 }
 
-// AppendEvent appends a new record to the CEL.
-func (c *CEL) AppendEvent(tpm io.ReadWriteCloser, pcr int, hashAlgos []crypto.Hash, event Content) error {
-	if len(hashAlgos) == 0 {
-		return fmt.Errorf("need to specify at least one hash algorithm")
-	}
+// generateDigestMap computes hashes with the given hash algos and the given event
+func generateDigestMap(hashAlgos []crypto.Hash, event Content) (map[crypto.Hash][]byte, error) {
 	digestsMap := make(map[crypto.Hash][]byte)
-
 	for _, hashAlgo := range hashAlgos {
 		digest, err := event.GenerateDigest(hashAlgo)
 		if err != nil {
-			return err
+			return digestsMap, err
 		}
 		digestsMap[hashAlgo] = digest
+	}
+	return digestsMap, nil
+}
+
+// AppendEvent appends a new record to the CEL. It's caller's responsibility to extend the hash
+// into the appropriate measurement register mapped to this canonical event log entry. The caller
+// must check the success of the extension; any failures to extend coupled with an AppendEvent
+// will lead to an inconsistent canonical eventlog and failure to replay/remote attestation.
+func (c *CEL) AppendEventRTMR(client configfsi.Client, rtmrIndex int, event Content) error {
+	digestsMap, err := generateDigestMap([]crypto.Hash{crypto.SHA384}, event)
+	if err != nil {
+		return err
+	}
+
+	eventTlv, err := event.GetTLV()
+	if err != nil {
+		return err
+	}
+
+	err = rtmr.ExtendDigestClient(client, rtmrIndex, digestsMap[crypto.SHA384])
+	if err != nil {
+		return err
+	}
 
-		tpm2Alg, err := tpm2.HashToAlgorithm(hashAlgo)
+	celrRTMR := Record{
+		RecNum:    uint64(len(c.Records)),
+		Index:     uint8(rtmrIndex) + 1, // RTMR[0] is CCEL[1]
+		Digests:   digestsMap,
+		Content:   eventTlv,
+		IndexType: RtmrTypeValue,
+	}
+
+	c.Records = append(c.Records, celrRTMR)
+	return nil
+}
+
+// AppendEvent appends a new record to the CEL. It's caller's responsibility to extend the hash
+// into the appropriate measurement register mapped to this canonical event log entry. The caller
+// must check the success of the extension; any failures to extend coupled with an AppendEvent
+// will lead to an inconsistent canonical eventlog and failure to replay/remote attestation.
+func (c *CEL) AppendEventPCR(tpm io.ReadWriteCloser, pcr int, hashAlgos []crypto.Hash, event Content) error {
```

Comment:
nit: could you also provide a wrapper func `AppendEvent` to call `AppendEventPCR`? `AppendEvent` is widely used by GAV testutils https://source.corp.google.com/piper///depot/google3/cloud/hosted/confidentialcomputing/testing/utilities/testutil.go;rcl=669028808;l=448. w/o this wrapper will lead to compile errors once this PR is imported back to G3.

Reply:
Added